src/states/elder/mod.rs

```diff
@@ -2158,9 +2147,15 @@ impl Approved for Elder {
                 trusted = Some(ps);
             }
         }
-        if validates(&trusted, &sec_info) || self.is_trusted(&sec_info)? {
-            let _ = self.add_new_section(&sec_info);
+        if !(validates(&trusted, &sec_info) || self.is_trusted(&sec_info)?) {
+            trace!(
```

Comment:
so this branch now becomes for logging purpose only?
shall we put it under a `debug` guard to save the computation?

Reply:
No need, we will have to change all that code to be secure with SMD, also unless profiling showed a problem we should not do these kind of optimization. Also log is used in release mode :-)